From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Li Hua Qian <huaqian.li@siemens.com>
Date: Fri, 1 Aug 2025 18:39:35 +0800
Subject: [PATCH] edk2-standalonemm-rpmb: Bump to 202505 and refresh crypto
 dependencies

Sync with latest upstream, requiring OpenSSL 3.4.1, MbedTLS 3.3.0, and
libspdm 3.6.0 per EDK2/edk2-platforms dependencies. Also pick the latest
brotli release with edk2 submodules.

Signed-off-by: Li Hua Qian <huaqian.li@siemens.com>
---
 ...05.bb => edk2-standalonemm-rpmb_202505.bb} | 30 ++++++++++++++-----
 1 file changed, 22 insertions(+), 8 deletions(-)
 rename recipes-bsp/edk2/{edk2-standalonemm-rpmb_202305.bb => edk2-standalonemm-rpmb_202505.bb} (62%)

diff --git a/recipes-bsp/edk2/edk2-standalonemm-rpmb_202305.bb b/recipes-bsp/edk2/edk2-standalonemm-rpmb_202505.bb
similarity index 62%
rename from recipes-bsp/edk2/edk2-standalonemm-rpmb_202305.bb
rename to recipes-bsp/edk2/edk2-standalonemm-rpmb_202505.bb
index 01dcbf9998be..4f80cc703df8 100644
--- a/recipes-bsp/edk2/edk2-standalonemm-rpmb_202305.bb
+++ b/recipes-bsp/edk2/edk2-standalonemm-rpmb_202505.bb
@@ -1,11 +1,12 @@
 #
 # CIP Core, generic profile
 #
-# Copyright (c) Siemens AG, 2022-2023
+# Copyright (c) Siemens AG, 2022-2025
 #
 # Authors:
 #  Sven Schultschik <sven.schultschik@siemens.com>
 #  Jan Kiszka <jan.kiszka@siemens.com>
+#  Li Hua Qian <huaqian.li@siemens.com>
 #
 # SPDX-License-Identifier: MIT
 #
@@ -21,22 +22,29 @@ SRC_URI = " \
     https://github.com/google/brotli/archive/${SRCREV-brotli}.tar.gz;name=brotli;subdir=${S} \
     https://github.com/MIPI-Alliance/public-mipi-sys-t/archive/${SRCREV-mipisyst}.tar.gz;name=mipisyst;subdir=${S} \
     https://github.com/openssl/openssl/archive/refs/tags/${PV-openssl}.tar.gz;name=openssl;subdir=${S} \
+    https://github.com/Mbed-TLS/mbedtls/archive/${PV-mbedtls}.tar.gz;name=mbedtls;subdir=${S} \
+    https://github.com/DMTF/libspdm/archive/refs/tags/${PV-libspdm}.tar.gz;name=libspdm;subdir=${S} \
     file://rules \
     "
-SRC_URI[sha256sum] = "5f6c18cf1068089d669fbe81dab2325f8bf7b1298b192c276490b65e2edbbd94"
-SRC_URI[edk2-platforms.sha256sum] = "31257160ac51a4a5f63db92d26482d6a005286ed040dafe89d01f0ee906b111e"
-SRC_URI[brotli.sha256sum] = "6d6cacce05086b7debe75127415ff9c3661849f564fe2f5f3b0383d48aa4ed77"
+SRC_URI[sha256sum] = "5f2b5e3a267230f82e4566592fd0bfac5e205ad90520b2c9bf80f575293b7015"
+SRC_URI[edk2-platforms.sha256sum] = "0a266a61732d8dbb95bd2a4c0bf64a91965728b9fcd567af6af0894f23da0f97"
+SRC_URI[brotli.sha256sum] = "aaa739962a45b508b2e783b915e6b2b57ed3b12bd4b0feac73acfb144dffa54f"
 SRC_URI[mipisyst.sha256sum] = "9fda3b9a78343ab2be6f06ce6396536e7e065abac29b47c8eb2e42cbb4c4f00b"
-SRC_URI[openssl.sha256sum] = "b1270f044e36452e15d1f2e18b702691a240b0445080282f2c7daaea8704ec5e"
+SRC_URI[openssl.sha256sum] = "d4b22527a645acf76b53e44487a8db687c6eed621d7246891d025e38ba8c9651"
+SRC_URI[mbedtls.sha256sum] = "a22ff38512697b9cd8472faa2ea2d35e320657f6d268def3a64765548b81c3ec"
+SRC_URI[libspdm.sha256sum] = "dfadf501d23c26041c921974971953a1d4a250ed6cd2679facb781471a5f944c"
 
 # according to edk2 submodules
-SRCREV-brotli = "f4153a09f87cbb9c826d8fc12c74642bb2d879ea"
+SRCREV-brotli = "ed738e842d2fbdf2d6459e39267a633c4a9b2f5d"
 SRCREV-mipisyst = "370b5944c046bab043dd8b133727b2135af7747a"
 
 # revision closest to edk2 release
-SRCREV-edk2-platforms = "b71f2bda9e4fc183068eef5d1d90a631181a2506"
+SRCREV-edk2-platforms = "cf63a7257a0a12044b4530904835427813855f4f"
+
+PV-openssl = "openssl-3.4.1"
+PV-mbedtls = "mbedtls-3.3.0"
+PV-libspdm = "3.6.0"
 
-PV-openssl = "OpenSSL_1_1_1t"
 
 DEBIAN_BUILD_DEPENDS = "bash, python3:native, dh-python, uuid-dev:native"
 
@@ -61,6 +69,12 @@ do_prepare_build() {
     rm -rf ${S}/edk2/CryptoPkg/Library/OpensslLib/openssl
     ln -s ../../../../openssl-${PV-openssl} ${S}/edk2/CryptoPkg/Library/OpensslLib/openssl
 
+    rm -rf ${S}/edk2/CryptoPkg/Library/MbedTlsLib/mbedtls
+    ln -s ../../../../mbedtls-${PV-mbedtls} ${S}/edk2/CryptoPkg/Library/MbedTlsLib/mbedtls
+
+    rm -rf ${S}/edk2/SecurityPkg/DeviceSecurity/SpdmLib/libspdm
+    ln -s ../../../../libspdm-${PV-libspdm} ${S}/edk2/SecurityPkg/DeviceSecurity/SpdmLib/libspdm
+
     echo "Build/MmStandaloneRpmb/RELEASE_GCC5/FV/BL32_AP_MM.fd /usr/lib/edk2/" > \
         ${S}/debian/edk2-standalonemm-rpmb.install
 }
